src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * @author Hang Zelin
+ * Main Programme to execute the Duke Project
+ * Duke will allow users to add three types of tasks: "todo" "deadline" "event".
+ * Duke also allow users to list all tasks, mark a task to be done if it is finished, and
+ * delete the task if the task is finished.
+ * You can also search a specific task by its date, keyword.
+ */
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.excpetions.DukeException;
+import duke.saveloadmanager.Storage;
+import duke.task.TaskList;
+import duke.uimanager.Ui;
+
+
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * initialize Ui, storage and load TaskLists from specific filePath for Duke
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Chooses a specific task to execute via tasks type and add to the tasklists.
+     * Every time an execution is done, the task will be stored to the local file called tasks.txt
+     * via Storage.
+     *
+     * @param operationType Type of the operation users input.
+     * @param task Task info users input.
+     * @param time Time info users input.
+     * @param index Index of the task users input.
+     */
+    public void operationForDuke(String operationType, String task, String time, int index) {
+        switch (operationType) {
+        case "bye": {
+            ui.goodbyeMessage();
+            break;
+        }
+        case "list": {
+            ui.printList(tasks);
+            break;
+        }
+        case "done": {
+            try {
+                tasks.detectIndex(index);
+                tasks.markDone(index);
+                ui.markDone(tasks.get(index).getTaskInfo());
+            } catch (DukeException e) {
+                e.printErrorMessage();
+            }
+            break;
+        }
+        case "delete": {
+            try {
+                tasks.detectIndex(index);
+                ui.delete(tasks.get(index).getTaskInfo(), tasks.size() - 1);
```

Comment:
perhaps can pass the handling of logical methods like delete and add to TaskList class?
Ui class sounds like it should just be handling the methods concerning interaction with user.